app/models/vegetable.rb

```diff
@@ -1,15 +1,11 @@
 class Vegetable < ApplicationRecord
   has_many :ratings
 
-  def as_json(options = {})
-    number_of_ratings = ratings.count
-    total = 0
-    
-    ratings.each do |rating|
-      total += rating.value
-    end
+  def average_rating
```

Comment:
Before continuing, check the commit history to see how I went about this change. I broke things into multiple commits so you could see the process of refactoring. The practice I like to follow is:

1. Write a test to validate the behavior (if there isn't one).
2. Once that test is passing, then proceed to refactoring the method.

That first step is very important. Without the tests case, you won't know if the refactoring changed the behavior.

Now, first things first, I felt like `as_json` was getting a bit busy. If we wanted to add more calculated fields, then the method would end up getting very unwieldy fairly quickly.

Second, while the previous way worked, it could end up being slow, depending on how many ratings a single vegetable has. Since this is used in the `/vegetables` endpoint, it has the potential to grow the amount of calculations exponentially. For each vegetable, we loop through it's ratings to calculate the sum, in order to get the average. If you're familiary with asymptotic notation, this would be in the O(n^2) category. We definitely don't want a lot of these hanging out in our codebase because they can slow things down pretty fast as our dataset grows!

In order to fix this, I used ActiveRecords `average` method. It let's the database do what it does best, and calculates the average for us.

Now! This code is _not_ perfect still, but I'm leaving that as homework for you. How else can we improve? What kind of new problem have I introduced?